typegraph/core/src/utils/fs_host.rs

```diff
@@ -201,11 +201,13 @@ pub fn make_absolute(path: &Path) -> Result<PathBuf, String> {
     }
 }
 
-pub fn hash_file(path: &Path) -> Result<String, String> {
+// TODO: use smaller buffer?
+pub fn hash_file(path: &Path) -> Result<(String, u32), String> {
```

Comment:
You mean like picking `min(actual-size, N)` first bytes?

Reply:
I meant, the whole file might not fit in memory. We can load it by a fixed size chunks for the hashing... Something with efficient disk reading and memory usage...